rosbag2_cpp/src/rosbag2_cpp/info.cpp

```
@@ -127,20 +171,112 @@ std::vector<std::shared_ptr<rosbag2_service_info_t>> Info::read_service_info(
-          break;
+      if (action_topic_type != TopicsInAction::Unknown) {
+        auto action_name = action_topic_name_to_action_name(bag_msg->topic_name);
+
+        // Handle action service event topic
+        switch (msg.event_type) {
+          case service_msgs::msg::ServiceEventInfo::REQUEST_SENT:
+          case service_msgs::msg::ServiceEventInfo::REQUEST_RECEIVED:
+            {
+              if (action_topic_type == TopicsInAction::SendGoalEvent) {
+                action_process_info[action_name]->
+                send_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
+              } else if (action_topic_type == TopicsInAction::GetResultEvent) {
+                action_process_info[action_name]->
+                get_result_service.request[msg.client_gid].emplace(msg.sequence_number);
+              } else {
+                // TopicsInAction::CancelGoalEvent
+                action_process_info[action_name]->
+                cancel_goal_service.request[msg.client_gid].emplace(msg.sequence_number);
+              }
+              break;
+            }
+          case service_msgs::msg::ServiceEventInfo::RESPONSE_SENT:
+          case service_msgs::msg::ServiceEventInfo::RESPONSE_RECEIVED:
+            {
+              if (action_topic_type == TopicsInAction::SendGoalEvent) {
+                action_process_info[action_name]->
+                send_goal_service.response[msg.client_gid].emplace(msg.sequence_number);
+              } else if (action_topic_type == TopicsInAction::GetResultEvent) {
+                action_process_info[action_name]->
+                get_result_service.response[msg.client_gid].emplace(msg.sequence_number);
+              } else {
+                // TopicsInAction::CancelGoalEvent
+                action_process_info[action_name]->
+                cancel_goal_service.response[msg.client_gid].emplace(msg.sequence_number);
+              }
+              break;
+            }
+        }
```

Please add the missing default case for the switch. We likely will need to throw an exception for the default case.